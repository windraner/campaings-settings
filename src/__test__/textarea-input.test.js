import React from 'react';
import TextareaInput from '../components/main-content-section/tab-block/common/textarea-input/textarea-input';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<TextareaInput />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<TextareaInput title="test1" value="test2" field="test3" stateHandler={function() {}} />);
    expect(wrapper.find('.campaigns-settings__input-title').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__input-title').text()).toBe('test1');
    expect(wrapper.find('.campaigns-settings__textarea-input').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__textarea-input').props().value).toBe('test2');
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<TextareaInput title="test1" value="test2" field="test3" stateHandler={onClickFunc} />);
    const input = wrapper.find('.campaigns-settings__textarea-input');
    input.simulate('change', { target: { value: 'foo' }});
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([['test3', 'foo']]);
  });
  it('matches the snapshot hide', () => {
    const tree = shallow(<TextareaInput title="test1" value="test2" field="test3" stateHandler={function() {}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
