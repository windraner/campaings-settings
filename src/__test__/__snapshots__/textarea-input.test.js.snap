// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextareaInput /> matches the snapshot hide 1`] = `
<Fragment>
  <label
    className="campaigns-settings__input-title"
  >
    test1
  </label>
  <textarea
    className="campaigns-settings__textarea-input d-block w-100 undefined"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value="test2"
  />
</Fragment>
`;
